runtime/tchannel_client.go

```diff
@@ -59,66 +59,92 @@ func NewTChannelClient(ch *tchannel.Channel, opt *TChannelClientOption) TChannel
 }
 
 func (c *tchannelClient) writeArgs(call *tchannel.OutboundCall, headers map[string]string, req RWTStruct) error {
-	writer, err := call.Arg2Writer()
+
+	bodyBuf := GetBuffer()
```

Comment:
Is the buffer here to make sure we don't write invalid partial RPC request? If so, I don't think we need it, because invalid partial request will fail the `ToWire` call [here](https://github.com/uber/zanzibar/blob/master/runtime/tchannel_helpers.go#L76) in the original implementation, and it will not be written to the writer.